### PR TITLE
Updated frontend for changes to dashboard API

### LINKED ISSUE #225: add program information (title) to student dashboard
front-end for #224

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #225 

#### What's this PR do?
Updates the frontend to use the updated dashboard API on the backend

#### Where should the reviewer start?
See `static/js/constants.js` to understand what changed, then look at `CourseList.js`

#### How should this be manually tested?
The dashboard should function correctly again, and now you should be able to see the program title in the table header.

### CODE REVIEW

File: static/js/components/CourseList.js [hunk 1]
```diff
@@ -10,47 +10,54 @@ class CourseList extends React.Component {
...
-          {makeCourseProgressDisplay(course, isTop, isBottom)}
-        </li>
-      </ul>;
+    let programs = dashboard.programs.map(program => {
+      let sortedCourses = _.sortBy(program.courses, 'position_in_program');
+
+      let table = sortedCourses.map((course, i) => {
+        // id number is not guaranteed to exist here so we need to use the whole
+        // object to test uniqueness
+        // TODO: fix this when we refactor
```
Comment (giocalitri) [inline static/js/components/CourseList.js]: is this still to be refactored or the comment is outdated?

**Comment by giocalitri** [comment]
As far as I can tell, the code looks fine.

**Comment by giocalitri** [comment]
functionality is good: the only problem I see with the UI is that in case of multiple programs, the look does not seem right:
screen shot 2016-04-29 at 9 47 43 pm

But I leave it to you if you want to fix it now or open an issue and merge.

File: static/js/components/CourseList.js (hunk 1 above)
Comment (noisecapella) [inline static/js/components/CourseList.js]: Comment is outdated, will fix

**Comment by noisecapella** [comment]
Force pushed by accident but the fixes should be there now

**Comment by giocalitri** [comment]
this is good to go 👍

**Pull request merged**